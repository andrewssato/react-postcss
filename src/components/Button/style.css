.button {
  border: 0;
  border-radius: 3px;
}