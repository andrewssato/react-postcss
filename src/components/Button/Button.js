
import React from 'react';
import PropTypes from 'prop-types';
import style from './style.css'

const Button = ({ type, children, theme }) => {
  return (
    <button type={type} className={theme.button}>
      { children || 'Button' }
    </button>
  );
};

Button.defaultProps = {
  theme: style,
  type: 'button',
};

Button.propTypes = {
  theme: PropTypes.shape({
    button: PropTypes.string.isRequired,
  }),
  type: PropTypes.string,
};

export default Button;