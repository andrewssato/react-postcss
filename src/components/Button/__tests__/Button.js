import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Button from '../';

describe('#Button', () => {
  // mock style postcss
  const theme = {button: 'button'};

  it('Deve renderizar Button', () => {
    const el = shallow(<Button theme={theme} />);
    expect(el.contains('Button')).toEqual(true);
    expect(el.find('button')).toHaveLength(1);
  });

  it('Deve escrever source do Button', () => {
    const tree = renderer.create(<Button theme={theme} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
