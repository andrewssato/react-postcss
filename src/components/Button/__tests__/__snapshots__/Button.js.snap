// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Button Deve escrever source do Button 1`] = `
<button
  className="button"
  type="button"
>
  Button
</button>
`;
